### ISSUE TITLE
Tag value for 'synced' was different from 'total'

### ISSUE THREAD

**Issue by zbiljic** [ethersphere/bee#1200]
#### Summary

The tag values after upload were invalid, and 'synced' was not same as 'total' or 'processed' counters.
This occurred during upload to `/bytes` endpoint, of a file ~40MB.

```json
{
    "uid": 2720375,
    "startedAt": "2021-02-05T15:12:27.495311+01:00",
    "total": 11164,
    "processed": 11164,
    "synced": 11166
}
```

Response for debug API endpoint for tags is:

```json
{
    "uid": 2720375,
    "startedAt": "2021-02-05T15:12:27.495311+01:00",
    "address": "",
    "total": 11164,
    "split": 11164,
    "seen": 955,
    "stored": 11164,
    "sent": 10376,
    "synced": 10211,
}
```

Same steps were done for multiple other files, on the same upload endpoint as well on others, with content ranging from `10MiB` to `100MiB`, and same issue did not occur.

In this case, most likely the `synced` counter was incremented more that it should have been.

This issue is mostly created to track any similar occurrence also.

#### Expected behavior

After upload is completed (and we know it is synced) all reported values for tag should have same value.

**Comment by Eknir** [comment]
@zbiljic , do you have any idea what is the cause of this?

**Comment by jpritikin** [comment]
I've seen the opposite problem. I uploaded, a [jiu jitsu video](https://gateway.ethswarm.org/files/18179c240373c5bb621e34cf608cdaeed72d76a6d5cf344d3308ffec7b6bc7b4) and [recipe for mocha pudding](https://gateway.ethswarm.org/files/9f74d494b4e8f44569f5357673cf5c5b4db8183b21be88483942734c30bbb2c1). The associated tags are,

```
{
  "uid": 5081206,
  "startedAt": "2021-02-18T12:34:50-05:00",
  "total": 33726,
  "processed": 33726,
  "synced": 0
}
```
and
```
{
  "uid": 1152502605,
  "startedAt": "2021-02-18T12:00:00-05:00",
  "total": 132225,
  "processed": 132225,
  "synced": 952
}
```

It seems like the 'synced' count gets reset to zero whenever I restart the bee node.

**Comment by Eknir** [comment]
Closed, in favor of #1371. Thanks for reporting!
